### ISSUE TITLE
elprep5: gVCF does not match BAM read pileup

### ISSUE THREAD

**Issue by splaisan** [ExaScience/elprep#61]
using elprep version 5.1.2 build he881be0_0 compiled with go1.17.6 from bioconda (https://anaconda.org/bioconda/elprep)
    ubuntu 20.04
    conda 4.12.0

I am very confused, I ran the pipeline on 7 chicken genomes sequenced with Novaseq PE150 and for all variants that I filter as being private for a single genome I also see the variant (with for some many read support) in several other samples using IGV and the elprep output BAM data.

Reads mapped with `BWA mem (-M -R ${Rstring})`

It looks like the gVCF produced by elprep is not reporting the correct variants (many missing calls).
I cannot say where the issue is but the BAM data does not agree with the separate gVCF outputs although the pileups are obvious.

my command was:

```
workdir=$PWD
cd ${wordkir}
mkdir -p tmpfiles elprep

# variables
reffolder=/data/biodata/references/galGal6a
ref=${reffolder}/Gallus_gallus.GRCg6a.dna.toplevel.elfasta
dbsnp=${reffolder}/ens_dbSNP_150/GRCg6a_dbSNP_150.elsites
pixeldist=100

# run loop over BWA mem raw mappings
for inbam in bwa_mappings/*_rawmappings.bam; do

pfx=$(basename ${inbam%_rawmappings.bam})
outbam=${workdir}/elprep/${pfx}_elprep_mappings.bam

elprep sfm ${inbam} ${outbam} \
  --optical-duplicates-pixel-distance ${pixeldist} \
  --mark-duplicates \
  --mark-optical-duplicates ${workdir}/elprep/${pfx}.output.metrics \
  --sorting-order coordinate \
  --bqsr ${workdir}/elprep/${pfx}_bqsr.output.recal \
  --known-sites ${dbsnp} \
  --reference ${ref}  \
  --haplotypecaller ${workdir}/elprep/${pfx}.g.vcf.gz \
  --reference-confidence GVCF \
  --nr-of-threads 80 \
  --timed \
  --log-path ${workdir}/elprep \
  --tmp-path ${workdir}/tmpfiles

done
```

Did you already have reports of such behavior or do you see something weird in my code?

Thanks for support

## random example where the T->C mutant is seen in 4 genomes but only called in #1 in the merged VCF derived from the 4 gVCF files.

![Screenshot 2022-03-24 at 09 42 15](https://user-images.githubusercontent.com/858516/159877206-2e4025b6-e45a-4c3a-a92a-efa7d5600028.png)

**Comment by caherzee** [comment]
Hi,

I must admit that elPrep has only been tested with human data. I do not particularly expect problems with chicken data, but I cannot be sure.

Another thing to keep in mind is that the elPrep haplotype caller was designed to produce identical results to the GATK haplotype caller (see our publications). We have had reports that elPrep did not detect specific variants, but that was the same for the aforementioned variant caller, and elPrep produced identical results.

There is of course also a chance that there is a bug in the elPrep haplotype caller. To test this, you could run the pipeline to produce the analysis-ready bam, but drop the --haplotypecaller option. Then you could use the analysis-ready bam with GATK and see if you see similar results or not.

If you find there is a bug, we appreciate hearing about it.

Thank you for testing elPrep!

Kind regards,
Charlotte

**Comment by splaisan** [comment]
Thanks Charlotte,

I have no hard evidence that elprep is specifically doing something wrong at this point, my data IS

After this issue, I tried other tools and got similar weird results to different extends.
I concluded that my genomes are too closely related and that the variant found by all pipelines are not true variants.

What remains weird is that all callers (freebayes deepvariant gatk/elprep and bcftools) call some genomes 0/0 at positions where other genomes are called 1/0. In such cases the 0/0 genomes do show about 4-5% of reads with the mutation when visualizing in IGV but are not called variant by any callers.

Probably 4-5% is less than the minimal threshold to call variants with all these callers?

Best
Stephane
